File: py/py_tests/algorithm_tests/breadth_first_search_test.py
""" This module holds unit tests for the breadth first search algorithm """
from py.data_structures.BinarySearchTree import BinarySearchTree
import unittest
from py.algorithms.breadth_first_search import breadth_first_search


class BreadthFirstSearchTests(unittest.TestCase):

    def test1(self):
        bst = BinarySearchTree(5)
        ints = [3, 5, 1, 2, -5, -9]
        for val in ints:
            bst.insertion(val)
        self.assertEqual(breadth_first_search(bst.root),
                         [[5], [3, 5], [1], [-5, 2], [-9]])


if __name__ == "__main__":
    unittest.main()

File: ts/src/tests/algorithm_tests/breadth_first_search.test.ts
import BinarySearchTree from '../../data structures/binary_search_tree';
import breadthFirstSearch from '../../algorithms/breadth_first_search';

test('test 1 for breadth first search algorithm', () => {
    const bst: BinarySearchTree = new BinarySearchTree(3);
    bst.insert(9);
    bst.insert(10);

    const output = [[3], [9], [10]];
    expect(breadthFirstSearch(bst.root)).toStrictEqual(output);
});

test('test 2 for breadth first search algorithm', () => {
    const bst: BinarySearchTree = new BinarySearchTree(3);

    const output = [[3]];
    expect(breadthFirstSearch(bst.root)).toStrictEqual(output);
});

test('test 3 for breadth first search algorithm', () => {
    const bst: BinarySearchTree = new BinarySearchTree(1235);
    bst.insert(50);
    bst.insert(95);
    bst.insert(3500);
    bst.insert(2400);
    bst.insert(2350);

    const output = [[1235], [50, 3500], [95, 2400], [2350]];
    console.log(breadthFirstSearch(bst.root));
    // expect(breadthFirstSearch(bst.root)).toStrictEqual(output);
});
